### PR TITLE
Convert `aten.bitwise_not` to `ttnn.bitwise_not`

### LINKED ISSUE #634: aten.bitwise_not.default

### PR DESCRIPTION
**Pull request by jdh8**
### Ticket
- Resolves #634
- Depends on tenstorrent/tt-metal#13579

### Problem description
`ttnn.bitwise_not` takes an extraneous `value: int`.  I'm fixing this in the kernel.
```
E       TypeError: __call__(): incompatible function arguments. The following argument types are supported:
E           1. (self: ttnn._ttnn.operations.unary.bitwise_not_t, input_tensor: ttnn._ttnn.tensor.Tensor, value: int, *, memory_config: Optional[ttnn._ttnn.tensor.MemoryConfig] = None, output_tensor: Optional[ttnn._ttnn.tensor.Tensor] = None, queue_id: int = 0) -> ttnn._ttnn.tensor.Tensor
E       
E       Invoked with: <ttnn._ttnn.operations.unary.bitwise_not_t object at 0x7f58fc4d5530>, ttnn.Tensor([[-81.00000, 66.00000,  ...,  0.00000,  0.00000],
E                    [58.00000, -75.00000,  ...,  0.00000,  0.00000],
E                    ...,
E                    [ 0.00000,  0.00000,  ...,  0.00000,  0.00000],
E                    [ 0.00000,  0.00000,  ...,  0.00000,  0.00000]], shape=Shape([4[32], 4[32]]), dtype=DataType::BFLOAT16, layout=Layout::TILE)

../tt-metal/ttnn/ttnn/decorators.py:326: TypeError
```

### What's changed
- [x] Lower `aten.bitwise_not` to `ttnn.bitwise_not`
- [x] Fix parameters of `ttnn.bitwise_not` in the kernel (either of the following)
  - tenstorrent/tt-metal#12902
  - tenstorrent/tt-metal#13579
- [ ] Passing test for the conversion

### CODE REVIEW

File: tests/lowering/eltwise/unary/test_bitwise_not.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        return torch.bitwise_not(input)
+
+
+@pytest.mark.parametrize(
+    "input_shape",
+    [(4, 4)],
+)
+def test_bitwise_not(device, input_shape):
+    m = BitwiseNotModule()
+    input = torch.randint(-256, 256, input_shape, dtype=torch.int32)
```
Comment (jdh8) [inline tests/lowering/eltwise/unary/test_bitwise_not.py:21]: Here is a problem.  I make this tensor in `torch.int32`, but the test complains that it gets a BFLOAT16 tensor.
```
E       RuntimeError: TT_FATAL @ /home/jdh8/tt-metal/ttnn/cpp/ttnn/operations/eltwise/unary/device/unary_device_operation.cpp:42: input_datatype == DataType::INT32
E       info:
E       Unsupported input data type '0' for UnaryOpType '63' (Bitwise operation).
E       backtrace:
E        --- /home/jdh8/tt-metal/ttnn/ttnn/_ttnn.so(+0x4933a9) [0x7f0db03ee3a9]
E        --- ttnn::operations::unary::UnaryDeviceOperation::validate_on_program_cache_miss(ttnn::operations::unary::operation_attributes_t const&, ttnn::operations::unary::tensor_args_t const&)
```

Previous error message also confirms this.

tenstorrent/tt-metal#12902 does not fix this error.

Comment (ayerofieiev-tt) [reply tests/lowering/eltwise/unary/test_bitwise_not.py:21]: @kevinwuTT ptal

Comment (jdh8) [reply tests/lowering/eltwise/unary/test_bitwise_not.py:21]: This error still persists after merging tenstorrent/tt-metal#13579.

Comment (jdh8) [reply tests/lowering/eltwise/unary/test_bitwise_not.py:21]: @pschang-phy, this was my failed attempt to lower `aten.bitwise_not`.  I just confirmed that this error still persists.  I hope this is helpful for your work on #634.

Comment (pschang-phy) [reply tests/lowering/eltwise/unary/test_bitwise_not.py:21]: Here is my try. It seems need to explicitly cast to **TtnnInt32**
```
cast_to_int32 = g.call_function(ttnn.typecast, args=(args[0], TtnnInt32()))
return g.call_function(ttnn.bitwise_not, args=(cast_to_int32,))
```

**Comment by pschang-phy** [comment]
@jdh8 I will take over this PR and push commit on it.

**Review by jerrysky3 — APPROVED** [review]

**Pull request merged**